jplag.frontend.java-1.13/src/main/java/jplag/java113/Parser.java

```diff
@@ -0,0 +1,311 @@
+				if (elseStatement.isPresent()) {
+					Position thenEnd = ifStmt.getThenStmt().getRange().map(r -> r.end).orElse(
+							elseStatement.get().getRange().map(r -> r.end).orElse(pos)
+					);
+					add(JavaTokenConstants.J_ELSE, thenEnd, "else");
+					dispatch(elseStatement.get());
+				}
+
+				add(JavaTokenConstants.J_IF_END, node.getRange().map(r -> r.end).orElse(pos), 1);
+			}
+
+			statement.ifWhileStmt($ -> add(JavaTokenConstants.J_WHILE_BEGIN, pos, "while"));
+			statement.ifReturnStmt($ -> add(JavaTokenConstants.J_RETURN, pos, "return"));
+			statement.ifYieldStmt($ -> add(JavaTokenConstants.J_YIELD, pos, "yield"));
+			statement.ifThrowStmt($ -> add(JavaTokenConstants.J_THROW, pos, "throw"));
+			if (statement.isTryStmt()) {
+				TryStmt tryStmt = statement.asTryStmt();
+				traversed = true;
+				if (tryStmt.getResources().isEmpty()) {
+					add(JavaTokenConstants.J_TRY_BEGIN, pos, "try");
+				} else {
+					add(JavaTokenConstants.J_TRY_WITH_RESOURCE, pos, "try");
+				}
+				tryStmt.getResources().forEach(this::dispatch);
+				NodeList<CatchClause> catchClauses = tryStmt.getCatchClauses();
+				catchClauses.forEach(this::dispatch);
+				Optional<BlockStmt> finallyBlock = tryStmt.getFinallyBlock();
+				if (finallyBlock.isPresent()) {
+					Optional<Range> finalRange = Optional.empty();
+					if (catchClauses.isNonEmpty()) {
+						finalRange = catchClauses.get(catchClauses.size() - 1).getRange();
+					}
+					var finalPos = finalRange.map((Range r) -> r.end).orElseGet(
+							() -> finallyBlock.get().getRange().map(r -> r.begin).orElse(pos)
+					);
+
+					add(JavaTokenConstants.J_FINALLY, finalPos, "finally");
+					dispatch(finallyBlock.get());
+				}
+			}
+			statement.ifExplicitConstructorInvocationStmt(invocationStmt -> add(
+					JavaTokenConstants.J_APPLY,
+					pos,
+					invocationStmt.isThis() ? "this" : "super"));
+		}
+
+		return traversed;
+	}
+
+	void dispatch(Node node) {
+		boolean traversed = this.beforeDispatch(node);
+
+		if (!traversed) {
+			node.getChildNodes().forEach(this::dispatch);
+
+			this.afterDispatch(node);
+		}
+	}
+}
+
+class JavaParserAdapter {
+	int parseFiles(File dir, File file, Parser parser) {
+		String filename = dir.toURI().relativize(file.toURI()).getPath();
+
+		CompilationUnit compilationUnit;
+		try {
+			compilationUnit = StaticJavaParser.parse(file);
+		} catch (FileNotFoundException | ParseProblemException e) {
+			e.printStackTrace();
+			return 1;
+
+		}
+
+		try {
+			new JavaParserWalker(parser, filename).dispatch(compilationUnit);
+		} finally {
+			parser.add(JavaTokenConstants.FILE_END.getValue(), filename, -1, -1, 1);
+		}
+
+		return 0;
+	}
+}
+
+public class Parser  extends jplag.Parser {
+	private jplag.Structure struct;
+
+	public static void main(String[] args) {
+		var parser = new Parser();
+		for (Token token : parser.parse(new File("/home/thomas/software/jplag/jplag.frontend.java-1.13/src/main/java/jplag/java113/"), new String[]{"Parser.java"}).tokens) {
```

Comment:
Is this path correct?

Reply:
Oh that was just some debug code, I'll remove it.